src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,18 +146,16 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
+                    var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationTokenSource.Token);
```

Comment:
We had a longer discussion with @danielmarbach about the currently proposed behavior. The first point we touched was the 'graceful shutdown' and aborting transactions when the outher `CancellationTokenSource` is set to cancelled. This currently because of the linkage between the outer source and a source created per message will abort all the in flight messages. Having that said, we considered the case where the transaction is not aborted when the source is cancelled. That guided the discussion towards FLR and the way they communicate the need of aborting the currently processed message. Because of the way SLR->FLR are configured, FLR cannot throw and simply Cancells the source. If we had SLR & FLR merged, this cancellation could be done with an exception, leaving only an additional cancellation token needed to be passed, to inform pipeline elements about the 'shutdown'.

All of the above, the FLR, SLR should not be done right now in this PR and unlinking the 'CancellationTokenSource' is what we need right now in this PR.